test/integration/common/StudioActions.kt

```diff
@@ -45,39 +45,71 @@ import java.nio.file.Path
 import java.nio.file.Paths
 import kotlin.test.assertEquals
 import kotlin.test.assertTrue
-import kotlin.test.fail
 import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
 import mu.KotlinLogging
 
 object StudioActions {
     private val LOGGER = KotlinLogging.logger {}
 
-    suspend fun clickIcon(composeRule: ComposeContentTestRule, icon: Icon, delayMillis: Int = Delays.RECOMPOSE) {
-        clickText(composeRule, icon.unicode, delayMillis)
+    suspend fun clickIcon(composeRule: ComposeContentTestRule, icon: Icon) {
+        clickText(composeRule, icon.unicode)
     }
 
-    suspend fun clickText(composeRule: ComposeContentTestRule, text: String, delayMillis: Int = Delays.RECOMPOSE) {
+    suspend fun clickText(composeRule: ComposeContentTestRule, text: String) {
+        waitUntilNodeWithTextIsClickable(composeRule, text)
         composeRule.onNodeWithText(text).performClick()
-        delayAndRecompose(composeRule, delayMillis)
+        delayAndRecompose(composeRule)
     }
 
-    fun clickAllInstancesOfText(composeRule: ComposeContentTestRule, text: String) {
+    suspend fun clickAllInstancesOfText(composeRule: ComposeContentTestRule, text: String) {
         val length = composeRule.onAllNodesWithText(text).fetchSemanticsNodes().size
         for (i in 0 until length) {
             composeRule.onAllNodesWithText(text)[i].performClick()
         }
+        delayAndRecompose(composeRule)
     }
 
-    fun clickAllInstancesOfIcon(composeRule: ComposeContentTestRule, icon: Icon) {
+    suspend fun clickAllInstancesOfIcon(composeRule: ComposeContentTestRule, icon: Icon) {
         clickAllInstancesOfText(composeRule, icon.unicode)
     }
 
-    fun assertNodeExistsWithText(composeRule: ComposeContentTestRule, text: String): SemanticsNodeInteraction {
-        return composeRule.onNodeWithText(text).assertExists()
+    suspend fun waitUntilNodeWithTextIsClickable(composeRule: ComposeContentTestRule, text: String) {
+        waitUntilAssertionIsTrue(composeRule) {
+            assertionDoesNotThrowException {
+                composeRule.onNodeWithText(text).assertHasClickAction()
+            }
+        }
+    }
+
+    suspend fun waitUntilNodeWithTextExists(composeRule: ComposeContentTestRule, text: String) {
+        waitUntilAssertionIsTrue(composeRule) {
+            assertionDoesNotThrowException {
+                composeRule.onNodeWithText(text).assertExists()
+            }
+        }
+    }
+
+    private fun assertionDoesNotThrowException(assertion: () -> Any): Boolean {
+        return try {
+            assertion()
+            true
+        } catch (e: Exception) {
+            false
+        }
     }
 
-    fun assertNodeNotExistsWithText(composeRule: ComposeContentTestRule, text: String) {
-        return composeRule.onNodeWithText(text).assertDoesNotExist()
+    suspend fun waitUntilAssertionIsTrue(composeRule: ComposeContentTestRule, assertion: () -> Boolean) {
```

Comment:
What this is really doing is just waiting for an arbitrary condition to become true. We should name the method as such, and have a separate method that runs an assertion function (that returns nothing, and is expected to throw if the assertion fails)

Reply:
Done!